src/twisted/conch/ssh/transport.py

```diff
@@ -726,11 +726,18 @@ def dataReceived(self, data):
         if not self.gotVersion:
             if self.buf.find(b'\n', self.buf.find(b'SSH-')) == -1:
                 return
+
+            # RFC 4253 section 4.2 ask for strict `\r\n` line ending.
+            # Here we are a bit more relaxed and accept implementations ending
+            # only in '\n'.
+            # https://tools.ietf.org/html/rfc4253#section-4.2
             lines = self.buf.split(b'\n')
             for p in lines:
                 if p.startswith(b'SSH-'):
                     self.gotVersion = True
-                    self.otherVersionString = p.strip()
+                    # Since the line was split on '\n' and most of the time
+                    # it uses '\r\n' we may get an extra '\r'.
+                    self.otherVersionString = p.rstrip('\r')
```

Comment:
Should this remove the last character if it happens to be `\r`?  A single instance of that is the only thing that have been part of the line delimiter.

Reply:
I don't understand the question.

If we have a line line, `SSH-bla-bla\r\rn' ... I think that it should just left a '\r' ... but I don't know if anybody is using that in production... and implementing that would require more work and tests.. so why bother? :)